app/v2/notifications/post_notifications.py

```diff
@@ -266,23 +271,26 @@ def save_email_to_queue(
     notification_id,
     form,
     notification_type,
-    api_key,
-    template,
+    api_key_id,
+    key_type,
+    template_id,
+    template_version,
     service_id,
     personalisation,
     document_download_count,
     reply_to_text=None
 ):
+    db.session.commit()
```

Comment:
is there any way we can check if the session has been used, and only trigger if that happens? I think if we want to make a pattern of this I'd like it to be as reusable and plug-and-play-able as possible. i've done a bit of digging but haven't got anywhere.

what i would like is us to be able to set a flag, such that if we then create a new connection, it raises an exception. Or something like that. So we're aware of what's happening.

I worry that we're just making our code a bit too brittle, but not in a way that will lead to errors. But someone will add `service.id` somewhere, and the code will work great but all the performance gains we get will disappear as it suddenly consumes a second connection.

Not sure if there's anything smart we can do with the sqlalchemy event hooks or anything 🤔

Reply:
also, at the very least, can you leave a comment here explaining why you're commiting. And leave a comment in `post_notifications` where save_email_to_queue is called explaining that it commits, and if we need stuff from the db we'll need to get it beforehand

Reply:
This is what https://github.com/alphagov/notifications-api/pull/2881 gives us. It:
- commits the session as soon as we no longer need it
- guarantees no subsequent database access, because `service` is no longer a SQLAlchemy object

Reply:
I'm not completely happy about having a commit here. But since this got in before the top down work Chris was doing, it ended up here. 
We can use db.session.expunge() which will throw an exception if you access a db object after the commit. However, we commit a notification in many places so introducing something like that will be a bit tricky.